src/libraries/System.Private.CoreLib/src/System.Private.CoreLib.Shared.projitems

```diff
@@ -1446,7 +1446,7 @@
   <ItemGroup Condition="'$(FeatureCominterop)' == 'true'">
     <Compile Include="$(MSBuildThisFileDirectory)System\Runtime\InteropServices\WindowsRuntime\EventRegistrationToken.cs" />
   </ItemGroup>
-  <ItemGroup Condition="'$(TargetsWindows)' == 'true' or '$(FeaturePal)'=='true'">
+  <ItemGroup Condition="'$(TargetsWindows)' == 'true' or '$(TargetsUnix)'=='true'">
```

Comment:
`FeaturePal` here was actually used to detect whether we are in CoreCLR build. I think you need define a new property in src\coreclr\src\system.private.CoreLib.csproj - e.g. `<FeatureCoreCLR>true</FeatureCoreCLR>` - and use it here as replacement of FeaturePal.

This is what's causing the build breaks in Mono.